### PR TITLE
Circuit for opcode TIMESTAMP

### PR DESCRIPTION
**Pull request by scroll-dev**
Implement the TimestampGadget for opcode TIMESTAMP.

Specs: [scroll-tech/zkevm-specs#31](https://github.com/appliedzkp/zkevm-specs/pull/105)

Other changes:

- Use `StackOnlyOpcode::<0>` for COINBASE bus mapping.
- Change mock block author address to a non-default value, so that it doesn't match the value in a default `BlockContext`.

### CODE REVIEW

**Review by ed255 — APPROVED** [review]
LGTM!  Please take a look at the note I wrote about using `cb.query_rlc()`.

Also let's wait for https://github.com/appliedzkp/zkevm-specs/pull/105 to be merged before merging this PR

File: zkevm-circuits/src/evm_circuit/execution/timestamp.rs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    value: RandomLinearCombination<F, N_BYTES_U64>,
+}
+
+impl<F: FieldExt> ExecutionGadget<F> for TimestampGadget<F> {
+    const NAME: &'static str = "TIMESTAMP";
+
+    const EXECUTION_STATE: ExecutionState = ExecutionState::TIMESTAMP;
+
+    fn configure(cb: &mut ConstraintBuilder<F>) -> Self {
+        let timestamp = array_init(|_| cb.query_cell());
```
[the remark this answers is not in the document]
Comment (z2trillion) [reply zkevm-circuits/src/evm_circuit/execution/timestamp.rs]: Thanks for the tip

**Comment by icemelon** [comment]
@ed255 Could you take a look again?

**Review by ed255 — APPROVED** [review]

**Comment by ed255** [comment]
Thanks for the work!

**Pull request merged**